### PR TITLE
Message PRs and stop release if test failed OPS-2362

### PR DESCRIPTION
**Pull request by nadeemshahzad**

### CODE REVIEW

File: edxpipelines/patterns/tasks/common.py [hunk 1]
```diff
@@ -172,7 +172,8 @@ def retrieve_artifact(artifact_location, job, dest=constants.ARTIFACT_PATH, runi
         stage=artifact_location.stage,
         job=artifact_location.job,
         src=src,
-        dest=dest
+        dest=dest,
+        runif=runif
```
Comment (feanil) [inline edxpipelines/patterns/tasks/common.py]: I believe you want this `runif` on the function for messaging PRs right?

https://github.com/edx/edx-gomatic/pull/375/files#diff-76c95fdd0bff54c5d08eb59951886d29R1494

File: edxpipelines/patterns/edxapp.py [hunk 2]
```diff
@@ -655,6 +656,27 @@ def generate_e2e_test_stage(pipeline, config):
...
+    head_ami_artifact = utils.ArtifactLocation(
+        "STAGE_edxapp_B",
+        constants.BUILD_AMI_STAGE_NAME,
+        constants.BUILD_AMI_JOB_NAME,
+        constants.BUILD_AMI_FILENAME,
+    )
+    for job in [e2e_tests, microsites_tests]:
+        tasks.generate_message_pull_requests_in_commit_range(
+            pipeline, job=job, org='edx', repo='edx-platform',
+            token='GITHUB_TOKEN', release_status=constants.ReleaseStatus.STAGED,
```
Comment (feanil) [inline edxpipelines/patterns/edxapp.py]: This isn't just that the artifacts are staged, you want to indicate that the e2e tests have failed on stage and provide a link to the pipeline.

See the undelying function: https://github.com/edx/tubular/blob/master/tubular/scripts/message_prs_in_range.py
and how it determines which message to send:  https://github.com/edx/tubular/blob/master/tubular/github_api.py

**Comment by coryleeio** [comment]
I believe @feanil's concern is addressed by the new PR, since he is out I'm going to dismiss his review so you can merge.  @fredsmith

**Comment by coryleeio** [comment]
This should be good to go now

**Pull request merged**